dandi/dandiapi.py

```diff
@@ -958,26 +980,21 @@ def wait_until_valid(self, max_time: float = 120) -> None:
         lgr.debug("Waiting for Dandiset %s to complete validation ...", self.identifier)
         start = time()
         while time() - start < max_time:
-            try:
-                r = self.client.get(f"{self.version_api_path}info/")
-            except HTTP404Error:
-                raise NotFoundError(
-                    f"No such version: {self.version_id!r} of Dandiset {self.identifier}"
-                )
-            if "status" not in r:
-                # Running against older version of dandi-api that doesn't
-                # validate
-                return
-            if r["status"] == "Valid":
+            v = self.get_version(self.version_id)
+            if v.status is VersionStatus.VALID:
                 return
             sleep(0.5)
         # TODO: Improve the presentation of the error messages
         about = {
-            "asset_validation_errors": r.get("asset_validation_errors"),
-            "version_validation_errors": r.get("version_validation_errors"),
+            "asset_validation_errors": [
+                e.json_dict() for e in v.asset_validation_errors
+            ],
+            "version_validation_errors": [
+                e.json_dict() for e in v.asset_validation_errors
```

Comment:
```suggestion
                e.json_dict() for e in v.version_validation_errors
```
as a duplication is evil.  What about 

```
about = {
    f"{scope}_validation_error" : [
        e.json_dict() for e in getattr(v, f"{scope}_validation_errors"))
    for scope in ('asset', 'version')
}
```
or alike altogether ?